core/src/main/java/org/mule/runtime/core/privileged/transaction/AbstractTransaction.java

```diff
@@ -122,6 +137,11 @@ public void rollback() throws TransactionException {
       setRollbackOnly();
       doRollback();
       fireNotification(new TransactionNotification(getId(), TRANSACTION_ROLLEDBACK, getApplicationName()));
+      if (rollbackAfterTimeout && timeoutReached()) {
```

Comment:
what if at the time of calling `rollback` it was within time, but at this call the timeout has passed?